### PR TITLE
Feat/3374  number of read messages

### PR DESCRIPTION
**Pull request by inkhey**
related to #3374 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/lib/core/event.py [hunk 1]
```diff
@@ -162,6 +162,16 @@ def get_paginated_messages_for_user(
         ).order_by(Message.event_id.desc())
         return get_page(query, per_page=count, page=page_token or False)
 
+    def get_nb_messages(
```
Comment (grignards) [inline backend/tracim_backend/lib/core/event.py:165]: `get_messages_count`? I'm unsure if there is a coding rule/a classic usage for this in the backend code.

File: backend/tracim_backend/models/context_models.py [hunk 2]
```diff
@@ -1562,3 +1571,15 @@ def __init__(self, page: Page) -> None:
         self.has_next = page.paging.has_next
         self.per_page = page.paging.per_page
         self.items = page
+
+
+class UserMessageSummary(object):
```
Comment (grignards) [inline backend/tracim_backend/models/context_models.py:1576]:
```suggestion
class UserMessagesSummary(object):
```

File: backend/tracim_backend/views/core_api/schemas.py [hunk 3]
```diff
@@ -1501,3 +1502,21 @@ class PathSuffixSchema(marshmallow.Schema):
         default="",
         example="/workspaces/1/notifications/activate",
     )
+
+
+class MessageSummaryQuerySchema(marshmallow.Schema):
```
Comment (grignards) [inline backend/tracim_backend/views/core_api/schemas.py:1507]:
```suggestion
class UserMessagesSummaryQuerySchema(marshmallow.Schema):
```

File: backend/tracim_backend/views/core_api/schemas.py [hunk 4]
```diff
@@ -1501,3 +1502,21 @@ class PathSuffixSchema(marshmallow.Schema):
...
+    """Possible query parameters for the GET messages summary endpoint."""
+
+    event_types = EventTypeListField()
+
+    @post_load
+    def message_summary_query(self, data: typing.Dict[str, typing.Any]) -> MessageSummaryQuery:
+        return MessageSummaryQuery(**data)
+
+
+class MessageSummarySchema(marshmallow.Schema):
```
Comment (grignards) [inline backend/tracim_backend/views/core_api/schemas.py:1517]:
```suggestion
class UserMessagesSummarySchema(marshmallow.Schema):
```

File: backend/tracim_backend/views/core_api/user_controller.py [hunk 5]
```diff
@@ -610,6 +614,36 @@ def get_user_messages(
             )
         )
 
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__USER_EVENT_ENDPOINTS])
+    @check_right(has_personal_access)
+    @hapic.input_path(UserIdPathSchema())
+    @hapic.input_query(MessageSummaryQuerySchema())
+    @hapic.output_body(MessageSummarySchema())
+    def get_user_messages_summary(self, context, request: TracimRequest, hapic_data: HapicData):
```
Comment (grignards) [inline backend/tracim_backend/views/core_api/user_controller.py:622]: Don't forget to annotate the return type.

File: backend/tracim_backend/lib/core/event.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/event.py]: done.

File: backend/tracim_backend/views/core_api/schemas.py (hunk 3 above)
Comment (inkhey) [reply backend/tracim_backend/views/core_api/schemas.py]: done.

File: backend/tracim_backend/views/core_api/schemas.py (hunk 4 above)
Comment (inkhey) [reply backend/tracim_backend/views/core_api/schemas.py]: done.

File: backend/tracim_backend/views/core_api/user_controller.py (hunk 5 above)
Comment (inkhey) [reply backend/tracim_backend/views/core_api/user_controller.py]: done.

File: backend/tracim_backend/models/context_models.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/models/context_models.py]: done.

File: backend/tracim_backend/models/context_models.py [hunk 6]
```diff
@@ -607,6 +607,15 @@ def __init__(
         self.event_types = event_types
 
 
+class MessageSummaryQuery(object):
```
Comment (grignards) [inline backend/tracim_backend/models/context_models.py:610]: I forgot this one…

```suggestion
class UserMessagesSummaryQuery(object):
```

Comment (inkhey) [reply backend/tracim_backend/models/context_models.py]: done.

**Pull request merged**